nengo/builder/ensemble.py

```diff
@@ -189,12 +189,11 @@ def build_ensemble(model, ens):
         encoders = npext.array(ens.encoders, min_dims=2, dtype=rc.float_dtype)
     if ens.normalize_encoders:
         encoders /= npext.norm(encoders, axis=1, keepdims=True)
-    n_nans = np.sum(np.isnan(encoders))
-    if n_nans > 0:
+    if np.any(np.isnan(encoders)):
         raise BuildError(
-            "%d encoders in %r are NaN. This usually means that you had zero-length "
+            "NaNs detected in %r encoders. This usually means that you had zero-length "
```

Comment:
Why remove the number of nans? Seems like useful information. I guess the downside is that the warning could show up multiple times for the same ensemble?

Reply:
Just didn't seem like useful information to me (I frequently want to know whether there are nans, but I have never cared whether there are 1 or 10, it's just a binary error). So removing that just allows us to keep the code simpler (and slightly faster, since `np.any` will be quicker than `np.sum`).  Relatively minor advantages, but slightly outweighs the even more minor benefit of knowing how many nans there are. Those were my thoughts anyway!

Reply:
My main reason for including it is that I can basically see two situations where you end up with zero length encoders. One is that you're doing random sampling poorly, like you're sampling a quantized range and you include zero in that range so you end up with 1 or 2 encoders that are zero. The other is that you are just setting all encoders to zero for some reason. But yeah, I guess in both of those situations, the underlying issue is one or more zeros, so it probably isn't important to differentiate :+1: